src/utils/data_reader.py

```
@@ -0,0 +1,136 @@
+        self,
+        source: str,
+        country: str,
+        start_date: datetime,
+        end_date: datetime,
+        parameters: Optional[List[str]] = None
+    ) -> pd.DataFrame:
+        if source not in ['openmeteo', 'nasapower']:
+            raise ValueError(f"Invalid weather source: {source}")
+        
+        files = self.manager.get_date_range_files(
+            source, start_date, end_date, country
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)